RA-13

maks9678 opened the pull request.

maks9678 committed:
RA-13 fragment_recipe.xml

maks9678 committed:
RA-12 correct

maks9678 committed:
RA-12 RecipesListFragment.kt,RecipeFragment.kt,MethodAdapter.kt

maks9678 committed:
RA-13 RecipesListFragment.kt,RecipeFragment.kt,MethodAdapter.kt

aGroupoid commented on app/src/main/java/com/example/androidsprint01/RecipeFragment.kt:
```diff
@@ -1,42 +1,6 @@
+/*
+val imageView: ImageView = findViewById(R.id.your_image_view)
+imageView.setColorFilter(Color.RED, PorterDuff.Mode.SRC_IN)
+изменение цвета иконки*/
```
Почему фрагмент заменён комментом? Удали этот файл

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipe/RecipeFragment.kt:
```diff
@@ -0,0 +1,79 @@
+        val ivRecipeImage: ImageView = binding.ivHeightRecipe
+
+
+        tvRecipeTitle.text = recipe?.title ?: "Рецепт"
```
Ссылаться всегда надо на строковые ресурсы, а не хардкожаные строки. Хардкод строки не могут поддерживать локализацию

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipe/RecipeFragment.kt:
```diff
@@ -0,0 +1,79 @@
+        val rvMethod: RecyclerView = binding.rvMethod
+        val methodAdapter = MethodAdapter(recipe?.method ?: emptyList())
+        rvMethod.adapter = methodAdapter
+        rvMethod.layoutManager = LinearLayoutManager(context)
```
Это статичное поле, которое со временм программы никак не меняется, его лучше вынести в xml файл

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipesList/RecipesListFragment.kt:
```diff
@@ -57,9 +70,9 @@ class RecipesListFragment : Fragment(R.layout.fragment_list_recipes) {
-        val recipesListAdapter =
+        val recipesListAdapter: RecipesListAdapter =
             RecipesListAdapter(BackendSingleton.getRecipesByCategoryId(id))
+        binding.rvListRecipes.layoutManager = LinearLayoutManager(context)
```
Эта строка всегда статичная, её нужно вынести в xml вёрстку, а отсюда убрать

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipesList/RecipesListFragment.kt:
```diff
@@ -57,9 +70,9 @@ class RecipesListFragment : Fragment(R.layout.fragment_list_recipes) {
-        binding.rvListRecipes.layoutManager = LinearLayoutManager(context)
-        val recipesListAdapter =
+        val recipesListAdapter: RecipesListAdapter =
             RecipesListAdapter(BackendSingleton.getRecipesByCategoryId(id))
```
Ресайклер у тебя не грузится, потому что ссылаешься на id фрагмента, а не на свою переменную categoryId

aGroupoid reviewed (changes requested):
С 11 задания ничего не работает. Просто молча не отправляй нерабочий код, описывай проблему. Иногда не успеваю полностью проверить работоспособность кода. Список не работает. Ты просто получаешь эти данные в аргументах, но никак не отрисовываешь их на макете. Через binding нужно проставлять вьюхам эти значения.
Пока исправь эти проблемы, потом посмотрим какие новые вылезут

maks9678 committed:
RA-13 RecipesListFragment.kt

aGroupoid commented:
> Ты просто получаешь эти данные в аргументах, но никак не отрисовываешь их на макете. Через binding нужно проставлять вьюхам эти значения.

Я выше писал, это не поправил. Данные из аргументов нужно присваивать к элементам view. Глянь еще раз задание, как там написано.

По Ra-13 - макет не соответсвует дизайну, во вьюхи шапки не переданы данные из аргументов. Каждый элемент recyclerView должен занимать wrap_content место. Шрифт, размеры текста и тд - всё должно соответствовать макету фигмы.

maks9678 committed:
RA-13 RecipesListFragment.kt

maks9678 committed:
RA-13 MethodAdapter and RecipeFragment
- Add prefix to steps in MethodAdapter
- Add padding for item_method.xml and item_ingredient.xml
- Load category image in RecipesListFragment

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipe/RecipeFragment.kt:
```diff
@@ -0,0 +1,90 @@
+
+    fun initRecycler() {
+        val dividerItem =
+            MaterialDividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL)
```
Девайдер нужно настроить по макету фигмы. Полоса заниемает не всю длину, у последнего элемента полосы нет

aGroupoid commented on app/src/main/res/layout/item_ingredient.xml:
```diff
@@ -0,0 +1,33 @@
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:background="@color/text_favorites_and_categories_and_fon_card">
```
У айтемов не должно быть такого бэкграунда. Сейчас скосы у каждого элемента. Чтоб было как на макете, нужно устанавливать его РесайклерВью

maks9678 committed:
RA-13 dividerItem

aGroupoid commented on app/src/main/java/com/example/androidsprint01/recipe/RecipeFragment.kt:
```diff
@@ -0,0 +1,94 @@
+            MaterialDividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL)
+        dividerItem.isLastItemDecorated = false
+        dividerItem.dividerColor =
+            ContextCompat.getColor(requireContext(), R.color.fon_navigation_bar)
```
Длину дивайдера настраивать через
divider.setDividerInsetStartResource
divider.setDividerInsetEndResource

aGroupoid commented on app/src/main/res/layout/item_method.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.cardview.widget.CardView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
```
Глянь как твои списки выглядят в темной теме. У карточки по умолчанию есть задний фон, его нужно отключить через app:cardBackgroundColor="@android:color/transparent">

maks9678 committed:
RA-13 dark theme

aGroupoid reviewed (approved):
В целом всё верно, импортируй только из фигмы оригинал изображений, а не обрезанные фотки из списка. Сделай это уже при выполнении следующего ПР
<img width="401" alt="Снимок экрана 2025-02-15 в 14 23 28" src="https://github.com/user-attachments/assets/54581b72-51d1-4a47-ba99-37c8c0cd070d" />

maks9678 committed:
RA-13 image
